### PR TITLE
Allow external LLVM and Clang folders.

### PR DESCRIPTION
**Pull request by dj2**
This CL updates the build files to allow setting the
CLSPV_LLVM_SOURCE_DIR and CLSPV_CLANG_SOURCE_DIR without specifying an
EXTERNAL_LLVM.

Because those directories maybe outside the clspv/ source tree we have
to provide the build directory when calling add_subdirectory.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -101,9 +107,8 @@ else()
   set(LLVM_TARGETS_TO_BUILD ""
       CACHE STRING "Semicolon-separated list of targets to build, or \"all\".")
 
-
   # Then pull in LLVM for building.
-  add_subdirectory(${CLSPV_LLVM_SOURCE_DIR} EXCLUDE_FROM_ALL)
```
Comment (alan-baker) [inline CMakeLists.txt]: Why is this necessary? It seems strange to add a directory in the build tree.

Comment (dj2) [reply CMakeLists.txt]: In my case I checkout third_party/clspv, third_party/clspv-llvm and third_party/clspv-clang. You can't add a subdirectory which is outside your subtree without providing the build directory. So, to keep things the same I set the build directory to what it would have been if it was in-subtree.
